### PR TITLE
Added a function to combine more metrics at once

### PR DESCRIPTION
**Pull request by Shawn-Guo-CN**
This update is to complete the second feature in issue #230 .

### CODE REVIEW

**Comment by colehurwitz** [comment]
Can you add some tests for this as well?

**Comment by colehurwitz** [comment]
Also, I will review this more when I get to be office.

**Comment by Shawn-Guo-CN** [comment]
> Can you add some tests for this as well?

Ummm, I didn't add any test because that I haven't fully understood how the current tests work. Maybe, we could discuss after your coming?

File: spiketoolkit/curation/threshold_metrics.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        return {
+            'snr': threshold_snr,
+            'num_spikes': threshold_num_spikes,
+            'firing_rate': threshold_firing_rate,
+            'isi_viol': threshold_isi_violations,
+            'presence_ratio': threshold_presence_ratio
+        }
+
+    @staticmethod
+    def build_curation_recording_need_list():
```
Comment (colehurwitz) [inline spiketoolkit/curation/threshold_metrics.py]: What is this part? I am a bit confused. Is this needed?

File: spiketoolkit/curation/threshold_metrics.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                elif mode == 'or':
+                    self._units_to_be_excluded = \
+                        self.union(self._units_to_be_excluded, _metric_units_to_be_excluded)
+                else:
+                    raise NotImplementedError('Mode %s is unknown' % mode)
+
+        self.exclude_units(self._units_to_be_excluded)
+        
+    @staticmethod
+    def build_curation_function_full_dict():
```
Comment (colehurwitz) [inline spiketoolkit/curation/threshold_metrics.py]: Can you explain this part?

**Comment by colehurwitz** [comment]
Looks good I think. Could you add a couple more tests for the other logical operations :D

**Comment by Shawn-Guo-CN** [comment]
This is just some notes to the future myself:

1. Consider implementing something like `soft` and `hard` curation in the parent class `ThresholdCurator`.
2. Take `spiketoolkit.validation.quality_metrics.py.compute_metrics()` in to account when designing the 'ThresholdMetrics' (the curator class).

**Comment by alejoe91** [comment]
Any update on this? I like the idea but I think it's outdated @Shawn-Guo-CN

**Comment by alejoe91** [comment]
Closing this for now..we can reopen it later!

**Pull request closed without merging**